corehq/pillows/domain.py

```diff
@@ -49,11 +49,11 @@ def calc_meta(self):
                                        "mapping": self.default_mapping})
 
     def change_transform(self, doc_dict):
-        doc_ret = copy.deepcopy(doc_dict)
         sub =  Subscription.objects.filter(
                 subscriber__domain=doc_dict['name'],
                 is_active=True)
         doc_dict['deployment'] = doc_dict.get('deployment', None) or {}
```

Comment:
That's what I thought too. I opted for leaving the side effect as-is, but I agree that your approach makes sense and would be better.